test/compilable/test16486.d

```diff
@@ -0,0 +1,36 @@
+// Example 1 - Simple case
+struct TestType(T, Q) { }
```

Comment:
The following doesn't work.
```suggestion
struct TestType(A, B) { }
alias TestAliasA(A, B) = TestType!(A, B);
alias TestAlias(T, Q) = TestAliasA!(T, Q);
```
Maybe you can replace the identifiers.

Reply:
Neither does this.
```suggestion
struct TestType(T, Q) { }
template TestAliasB(T, Q)
{
    alias A = T;
    alias TestAliasB = TestType!(A, Q);
}
alias TestAlias(T, Q) = TestAliasB!(T, Q);
```

Reply:
The second is still a little obscure to me. I'll try to understand it again.

For the first one, I understand why it makes sense to work. But, trying to implement it, here are the consequences. At some point, we reach the declaration `TestAliasA(A, B)` and this has type
`TestType!(A, B)`. This I think is like an incomplete type. Incomplete because of the `A`, `B`, which are identifiers, and their resolution depends on the scope. This is resolved depending on where it's used.
It seems I could replace `A`, `B` with whatever were the initial parameters. The problem with that is that we have now changed the type of the declaration `TestAliasA`. Code-wise, the consequence of that is that if we reach a template and its argument names don't match exactly the initial ones, we have to create a new type. I think that no matter how many alias indirections we want, we can just make a copy only in the last (the one that resolves to a non-alias template instance).
Is that acceptable?

Reply:
Now it seems that both pass. That was too easy to be correct though...

Reply:
Too simplistic. This doesn't wrok:
```suggestion
alias TestAlias(T, Q) = TestAliasA!(Q, T);
```
You should follow the parameters throughout, don't count on the order.

Reply:
Indeed. I realized, after the commit, by testing something like this, that some "following cleverness" should be added. Thanks though! Actually, for such parameters it is not difficult. But I didn't update it since then I thought about cases like:

```
struct TestType(A, B) { }
alias TestAlias(A: A*, B) = TestType!(A*, B);
alias TestAlias2(T, Q) = TestAlias!(T*, Q);
```
Something like that. Here, `A*` is different from `T` (the initial argument) and so it will be replaced
by `T`.
Meaning, "following parameters" is not just replacing identifiers. Still, it seems feasible because the ways that you can pass template parameters, however complicated they might be, all "look" like a declaration (they may even be parsed like that, I haven't looked). That is, they have only one identifier and that is the only you need to find and replace. But I better look at it tomorrow with a clear head.
Lastly, the follow-through, as said before, at least as I'm thinking it now, requires copies all the way.